### ISSUE TITLE
The maximum reward appears in the middle

### ISSUE THREAD

**Issue by xtc002007** [Unity-Technologies/ml-agents#3458]
Hi!ALL:

I encountered a problem during training. The reward can reach the maximum in the middle, but then the return will gradually decrease. What is the reason for this? Please help analyze it, thank you!

![2612](https://user-images.githubusercontent.com/14798607/74601473-c7e4b000-50d9-11ea-80ba-16b7534d1d64.png)

My code is:
    public override void CollectObservations()
    {
        AddVectorObs(meney);
        AddVectorObs(element1);
        AddVectorObs(element2);
        AddVectorObs(element3);
        AddVectorObs(weapon);
    }

    int coastNumber = 0;
    int lastCoastNumber = 0;
    int logNumber = 0;
    public override void AgentAction(float[] vectorAction)
    {
        bool b_money        = (vectorAction[0] == 1) ? true : false;
        int element         = (int)vectorAction[1];
        bool b_weapon       = (vectorAction[2] == 1) ? true : false;

        if (b_money)
        {
            meney += 10;
        }
        if (b_weapon)
        {
            if (meney >= 60 && element1 > 2 && element2 > 2 && element3 > 2)
            {
                //Debug.Log("------------------ make weapon ok -------------------");
                meney -= 20;
                element1 -= 2;
                element2 -= 2;
                element3 -= 2;
                weapon++;
            }
        }
        switch (element)
        {
            case 0:
                break;
            case 1:
                if (meney >= 100)
                {
                    meney -= 2;
                    element1++;
                }
                break;
            case 2:
                if (meney >= 100)
                {
                    meney -= 5;
                    element2++;
                }
                break;
            case 3:
                if (meney >= 100)
                {
                    meney -= 10;
                    element3++;
                }
                break;
        }
        AddReward(-0.01f);

        if (weapon > 5)
        {
            Debug.Log("------------------ weapon > " + weapon + " -------------------");
            SetReward(1f);
            lastCoastNumber = coastNumber;
            Done();
        }
    }

**Comment by harperj** [comment]
Hi @xtc002007 -- unfortunately our team doesn't have the resources to help debug training issues with custom training environments.  Anecdotally, many of us have had certain reward functions and environments regress during the training process.

You might find more support from the community on our forums: https://forum.unity.com/forums/ml-agents.453/

**Comment by xtc002007** [comment]
Thank you for your answer. Can I turn on debug mode during training and output more information? In order to observe some details in the process

**Comment by xtc002007** [comment]
@harperj    Thank you for your answer. Can I turn on debug mode during training and output more information? In order to observe some details in the process

**Comment by harperj** [comment]
@xtc002007 I'm doubtful you'll get much more that is useful from the debug flag.  I'd recommend making sure your observations cover all of the important information the agents need for decision making and that you give reward for any incremental progress.

Stopping training and loading up the model into your game to look at how the agent behaves might also be useful.

**Comment by xtc002007** [comment]
@harperj 
I currently have two questions:
1  How to turn on debug flag in python training program?
2  In the current training process, the continuous output of vectorAction values will be very concentrated, for example, vectorAction [0] will be the same value continuously for a period of time, and other vectorAction [n] will remain the same, even an episode is such. I learned earlier that at the beginning of training, the value of each vectorAction [n] was equal. Could this be caused by a configuration file? My configuration file is as follows:
default:
    trainer: ppo
    batch_size: 1024
    beta: 5.0e-3
    buffer_size: 10240
    epsilon: 0.2
    hidden_units: 128
    lambd: 0.95
    learning_rate: 3.0e-4
    learning_rate_schedule: linear
    max_steps: 5.0e5
    memory_size: 256
    normalize: false
    num_epoch: 3
    num_layers: 2
    time_horizon: 64
    sequence_length: 64
    summary_freq: 10000
    use_recurrent: false
    vis_encode_type: simple
    reward_signals:
        extrinsic:
            strength: 1.0
            gamma: 0.99
            
War03:
    batch_size: 128
    buffer_size: 4096
    epsilon: 0.3
    beta: 5.0e-2

**Comment by xtc002007** [comment]
@harperj 
I have solved this problem, mainly because the input values are not normalized. Thank you

**Comment by harperj** [comment]
Hi @xtc002007 -- glad you were able to figure out the problem!  I'll close this issue now, but feel free to open a new issue if you run into any further issues.
